File: Medium/Python/74. Search a 2D Matrix.py
class Solution(object):
    def searchMatrix(self, matrix, target):
        r = len(matrix)
        c = len(matrix[0])
        i = 0
        j = c - 1

        while 0 <= i < r and 0 <= j < c:
            if matrix[i][j] == target:
                return True
            elif target < matrix[i][j]:
                j -= 1
            else:
                i += 1

        return False

File: Medium/JavaScript/74. Search a 2D Matrix.js
var searchMatrix = function(matrix, target) {
    let r = matrix.length;
    let c = matrix[0].length;
    let i = 0;
    let j = c - 1;

    while(0 <= i && i < r && 0 <= j && j < c){
        if(matrix[i][j] == target)
            return true;
        else if(target < matrix[i][j])
            j--;
        else
            i++;
    }

    return false;
};
